### PR TITLE
refactor: ktor client 2.0

### LINKED ISSUE #313: Ktor 2.0.0-beta-1 is released
It would be better to support Ktor 2 beta with a beta sdk. 🙇

### PR DESCRIPTION
**Pull request by Aallam**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no    
| BC breaks?        | yes
| Related Issue     | Fix #313 
| Need Doc update   | yes

## Describe your change

* Updates Ktor to 2.x
* Migrate from Ktor 1.x features to Ktor 2.x plugins

### CODE REVIEW

**Comment by b95505017** [comment]
@Aallam Is it possible to merge it and release a beta version?

**Comment by Aallam** [comment]
> @Aallam Is it possible to merge it and release a beta version?

We are looking into it. Since this is a major Ktor release, we have a few things to consider before going any further.

**Comment by b95505017** [comment]
@Aallam FYI [Ktor 2.0.0 released](https://github.com/ktorio/ktor/releases/tag/2.0.0)

**Review by VladislavFitz — APPROVED** [review]
LGTM, just one comment by curiosity :)

File: client/src/commonMain/kotlin/com/algolia/search/transport/internal/HttpRequestBuilder.kt [hunk 1]
```diff
@@ -8,17 +8,18 @@ import com.algolia.search.transport.RequestOptions
 import io.ktor.client.request.HttpRequestBuilder
 import io.ktor.client.request.header
 import io.ktor.client.request.parameter
+import io.ktor.client.request.setBody
 
-internal fun HttpRequestBuilder.setApplicationId(applicationID: ApplicationID?) {
+internal fun HttpRequestBuilder.applicationId(applicationID: ApplicationID?) {
```
[the remark this answers is not in the document]
Comment (Aallam) [reply client/src/commonMain/kotlin/com/algolia/search/transport/internal/HttpRequestBuilder.kt:13]: Those functions don't follow the code style of the rest of the code base, that's all :D

**Pull request merged**